tests/forge/MixedSwapCallParameters.t.sol

```diff
@@ -0,0 +1,238 @@
+        require(success, "call failed");
+        assertEq(balanceOfBefore - USDC.balanceOf(from), 58607323117);
+        assertEq(ethBalanceOfBefore - address(from).balance, 198890409591425744);
+        assertEq(nft.balanceOf(RECIPIENT), 1);
+        assertEq(address(router).balance, 0);
+    }
+
+    function testMixedERC20ForLooksRareAndSeaportNFTs() public {
+        MethodParameters memory params = readFixture(json, "._ERC20_FOR_1_LOOKSRARE_NFT_2_SEAPORT_NFTS");
+
+        ERC721 nft = ERC721(0x5180db8F5c931aaE63c74266b211F580155ecac8);
+        vm.createSelectFork(vm.envString("FORK_URL"), 15360000);
+        vm.startPrank(from);
+
+        (router, permit2) = deployFixtureMainnetConfig();
+        vm.deal(from, BALANCE);
+
+        deal(address(USDC), from, BALANCE);
+        USDC.approve(address(permit2), BALANCE);
+        permit2.approve(address(USDC), address(router), uint160(BALANCE), uint48(block.timestamp + 1000));
+        assertEq(USDC.balanceOf(from), BALANCE);
+
+        uint256 balanceOfBefore = USDC.balanceOf(from);
+        uint256 ethBalanceOfBefore = address(from).balance;
+        (bool success,) = address(router).call{value: params.value}(params.data);
+        require(success, "call failed");
+        assertEq(balanceOfBefore - USDC.balanceOf(from), 156678862725);
+        assertEq(ethBalanceOfBefore - address(from).balance, 0);
+        assertEq(nft.balanceOf(RECIPIENT), 3);
+        assertEq(address(router).balance, 0);
+    }
+
+    function testMixedERC20AndETHForLooksRareAndSeaportNFTs() public {
+        MethodParameters memory params = readFixture(json, "._ERC20_AND_ETH_FOR_1_LOOKSRARE_NFT_2_SEAPORT_NFTS");
+
+        ERC721 nft = ERC721(0x5180db8F5c931aaE63c74266b211F580155ecac8);
+        vm.createSelectFork(vm.envString("FORK_URL"), 15360000);
+        vm.startPrank(from);
+
+        (router, permit2) = deployFixtureMainnetConfig();
+        vm.deal(from, BALANCE);
+
+        deal(address(USDC), from, BALANCE);
+        USDC.approve(address(permit2), BALANCE);
+        permit2.approve(address(USDC), address(router), uint160(BALANCE), uint48(block.timestamp + 1000));
+        assertEq(USDC.balanceOf(from), BALANCE);
+
+        uint256 balanceOfBefore = USDC.balanceOf(from);
+        uint256 ethBalanceOfBefore = address(from).balance;
+        (bool success,) = address(router).call{value: params.value}(params.data);
+        require(success, "call failed");
+        assertEq(balanceOfBefore - USDC.balanceOf(from), 58973906433);
+        assertEq(ethBalanceOfBefore - address(from).balance, 53000000000000000000);
+        assertEq(nft.balanceOf(RECIPIENT), 3);
+        assertEq(address(router).balance, 0);
+    }
+
+    function testMixedERC20ForNFTRevertsOnFailedUniswapSwap() public {
+        MethodParameters memory params = readFixture(json, "._ERC20_AND_ETH_FOR_1_LOOKSRARE_NFT_2_SEAPORT_NFTS");
+
+        ERC721 nft = ERC721(0x5180db8F5c931aaE63c74266b211F580155ecac8);
```

Comment:
given these lines are repeated in all tests you could factor them out into the `setUp()` function?